packages/schematics/update/update/npm.ts

```diff
@@ -33,15 +33,21 @@ function _readNpmRc(): Observable<{ [key: string]: string }> {
     // TODO: have a way to read options without using fs directly.
     const path = require('path');
     const fs = require('fs');
+    const perProjectNpmrc = path.resolve('.npmrc');
 
     let npmrc = '';
-    if (process.platform === 'win32') {
-      if (process.env.LOCALAPPDATA) {
-        npmrc = fs.readFileSync(path.join(process.env.LOCALAPPDATA, '.npmrc')).toString('utf-8');
-      }
+
+    if (fs.existsSync(perProjectNpmrc)) {
+      npmrc = fs.readFileSync(perProjectNpmrc).toString('utf-8');
     } else {
```

Comment:
You should parse per project .npmrc and user .nmprc because the repository url can be in the per project file and credentials in the user file

Reply:
True.

However if we are too picky here we enter the pain zone. First, then we need to support the global npmrc + userconfig flag / env variable as well. Second, we need to properly resolve the collisions between _authToken and _auth (because both could be present and we need to pick). Finally, the current approach is reading npmrc every time the package info is getting resolved which is not a standard npm behavior. Etc.

I'm not yet talking about having two different ways of getting NPM parameters.

The whole approach is not perfect and this PR does not pretend to be a fully working solution; at least it would be way more helpful than the current approach. Probably one needs to deeply rethink the concept.

Reply:
The private repos are nearly never used open source, hence no auth is required. Also if you change the order of resolving those files then you go in clear conflict with Npm itself

Reply:
#10624 is regarding an issue with fontawesome pro, so I guess it's used in open source. and npm don't need authentication nor .npmrc.

Reply:
https://fontawesome.com/pro - 60$ per license for 5 seats. What open source projects are you talking about?

Reply:
OK, I have more arguments, but the discussion went away from the issue. What we discuss is only the cases when there is a project-specific .npmrc which is not containing `_auth` or `_authToken`, but there is one in some of the .npmrc above right?

Reply:
Exactly, in that case, your PR doesn't fix the issue I guess